### PR TITLE
Dịch "gd_vn" - Phần 8

### LINKED ISSUE #1866: Dịch "gd_vn" - Phần 8
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_optimization/gd_vn.md#L548-L602

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tamio96**
"Assuming that for both the natural scale is in meters we have a terrible mismatch in parameterizations."
câu này mình không hiểu rõ lắm :\

## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1866 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_optimization/gd_vn.md [hunk 1]
```diff
@@ -562,8 +562,12 @@ Moreover, estimates for the main diagonal elements are what drives some of the i
 This leads to update algorithms of the form
 -->
 
-*dịch đoạn phía trên*
-
+Không có gì ngạc nhiên khi việc tính toán và lưu trữ toàn bộ ma trận Hessian là rất tốn kém.
+Do đó ta cần tìm kiếm một phương pháp thay thế.
+Một cách để cải thiện vấn đề này là tránh việc tính toán toàn bộ ma trận Hessian, chỉ tính toán trên các giá trị thuộc đường chéo.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:567]:
```suggestion
Một cách để cải thiện vấn đề này là tránh tính toán toàn bộ ma trận Hessian, chỉ tính toán các giá trị thuộc *đường chéo*.
```

File: chapter_optimization/gd_vn.md [hunk 2]
```diff
@@ -562,8 +562,12 @@ Moreover, estimates for the main diagonal elements are what drives some of the i
...
+Mặc dù cách trên không tốt bằng phương pháp Newton hoàn chỉnh nhưng vẫn tốt hơn nhiều so với không dùng qua.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:568]:
```suggestion
Mặc dù cách trên không tốt bằng phương pháp Newton hoàn chỉnh nhưng vẫn tốt hơn nhiều so với không sử dụng.
```

File: chapter_optimization/gd_vn.md [hunk 3]
```diff
@@ -562,8 +562,12 @@ Moreover, estimates for the main diagonal elements are what drives some of the i
...
+Hơn nữa, ước lượng các giá trị đường chéo chính chính là thứ thúc đẩy sự đổi mới trong các thuật toán tối ưu hóa hạ gradient ngẫu nhiên.
+Nó khiến các thuật toán cập nhật có dạng
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:570]:
```suggestion
Thuật toán cập nhật sẽ có dạng
```

File: chapter_optimization/gd_vn.md [hunk 4]
```diff
@@ -575,29 +579,36 @@ Using preconditioning removes this.
 Effectively preconditioning with gradient descent amounts to selecting a different learning rate for each coordinate. 
 -->
 
-*dịch đoạn phía trên*
+Để thấy tại sao điều này có thể là một ý tưởng tốt, ta ví dụ có hai biến số biểu thị chiều cao, một biến với đơn vị mm, biến còn lại với đơn vị km.
+Giả định rằng với cả hai thang đo tự nhiên tính bằng mét, chúng ta đều có sự sai khớp lớn trong việc tham số hóa.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:583]:
```suggestion
Với cả hai đơn vị đo, khi quy đổi ra mét, chúng ta đều có sự sai lệch lớn trong việc tham số hóa.
```
theo mình hiểu là vầy. Chờ reviewers khác nữa.

File: chapter_optimization/gd_vn.md [hunk 5]
```diff
@@ -575,29 +579,36 @@ Using preconditioning removes this.
...
+Sử dụng tiền đề sẽ xóa bỏ điều này.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:584]:
```suggestion
Sử dụng tiền đề sẽ loại bỏ điều này.
```

File: chapter_optimization/gd_vn.md [hunk 6]
```diff
@@ -575,29 +579,36 @@ Using preconditioning removes this.
...
+Tiền đề hiệu quả cùng hạ gradient có ý nghĩa giúp chọn ra các tốc độ học khác nhau cho từng trục tọa độ.
 
 <!--
 ### Gradient Descent with Line Search
 -->
 
-### *dịch đoạn phía trên*
+### Hạ gradient cùng tìm kiếm đường thẳng
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:591]:
```suggestion
### Hạ gradient cùng tìm kiếm đường
```

File: chapter_optimization/gd_vn.md [hunk 7]
```diff
@@ -575,29 +579,36 @@ Using preconditioning removes this.
...
 
 <!--
 One of the key problems in gradient descent was that we might overshoot the goal or make insufficient progress.
 A simple fix for the problem is to use line search in conjunction with gradient descent.
 That is, we use the direction given by $\nabla f(\mathbf{x})$ and then perform binary search as to which step length $\eta$ minimizes $f(\mathbf{x} - \eta \nabla f(\mathbf{x}))$. 
 -->
 
-*dịch đoạn phía trên*
+Một trong những vấn đề chính của hạ gradient là chúng ta có thể vượt lố khỏi mục tiêu hoặc không đạt đủ sự tiến bộ.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:599]:
```suggestion
Một trong những vấn đề chính của hạ gradient là chúng ta có thể vượt quá khỏi mục tiêu hoặc không đạt đủ sự tiến bộ.
```

File: chapter_optimization/gd_vn.md [hunk 8]
```diff
@@ -575,29 +579,36 @@ Using preconditioning removes this.
...
+Có một cách khắc phục đơn giản cho vấn đề này là sử dụng tìm kiếm đường thẳng kết hợp với hạ gradient.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:600]:
```suggestion
Có một cách khắc phục đơn giản cho vấn đề này là sử dụng tìm kiếm đường kết hợp với hạ gradient.
```

File: chapter_optimization/gd_vn.md [hunk 9]
```diff
@@ -575,29 +579,36 @@ Using preconditioning removes this.
...
 
 <!--
 This algorithm converges rapidly (for an analysis and proof see e.g., :cite:`Boyd.Vandenberghe.2004`).
 However, for the purpose of deep learning this is not quite so feasible, since each step of the line search would require us to evaluate the objective function on the entire dataset.
 This is way too costly to accomplish. 
 -->
 
-*dịch đoạn phía trên*
+Thuật toán này sẽ hội tụ nhanh chóng (xem phân tích và chứng minh ở :cite:`Boyd.Vandenberghe.2004`).
+Tuy nhiên, đối với mục đích của học sâu thì nó không quá khả thi do mỗi bước của tìm kiếm đường thẳng sẽ yêu cầu chúng ta ước lượng hàm mục tiêu trên toàn bộ tập dữ liệu.
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:610]:
```suggestion
Tuy nhiên, đối với mục đích của học sâu thì nó không quá khả thi do mỗi bước của tìm kiếm đường sẽ yêu cầu chúng ta ước lượng hàm mục tiêu trên toàn bộ tập dữ liệu.
```

File: chapter_optimization/gd_vn.md [hunk 10]
```diff
@@ -551,7 +551,7 @@ It comes down to ensuring that $f$ does not have any "surprising" properties in
 ### Preconditioning
 -->
 
-### *dịch đoạn phía trên*
+### Tiền đề
```
Comment (cuongvng) [inline chapter_optimization/gd_vn.md:554]: #1941

File: chapter_optimization/gd_vn.md (hunk 6 above)
Comment (tamio96) [reply chapter_optimization/gd_vn.md:591]: cái này cần phải bàn thêm

File: chapter_optimization/gd_vn.md (hunk 1 above)
Comment (tamio96) [reply chapter_optimization/gd_vn.md:567]: các cmt còn lại chủ yếu là cách diễn đạt khác nhau của 2 người chứ ko ảnh hưởng tới nội dung nên mình sẽ không commit

File: chapter_optimization/gd_vn.md (hunk 10 above)
Comment (duythanhvn) [inline chapter_optimization/gd_vn.md:554]:
```suggestion
### Tiền Điều kiện
```

File: chapter_optimization/gd_vn.md (hunk 6 above)
Comment (duythanhvn) [reply chapter_optimization/gd_vn.md:591]: @cuongvng @tamio96 hai bạn thảo luận nốt chỗ này trong ngày 01/07 giúp anh nhé.

Comment (cuongvng) [reply chapter_optimization/gd_vn.md:591]: em đang chờ các ý kiến khác. #1982 nhé.
